src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1098,8 +1098,6 @@ protected BoundExpression BindInferredVariableInitializer(BindingDiagnosticBag d
                 }
             }
 
-            CheckRestrictedTypeInAsyncMethod(this.ContainingMemberOrLambda, declTypeOpt.Type, localDiagnostics, typeSyntax);
```

Comment:
>CheckRestrictedTypeInAsyncMethod [](http://example.com/codeflow?start=12&length=32)

Are we still reporting an error for a `ref struct` local declared in an `async` method when compiling with earlier language version? For instance, compiling the following with `-langversion:12`.
```csharp
using System.Threading.Tasks;
ref struct R { }
class C
{
    async Task M()
    {
        R r = default;
        await Task.FromResult(false);
    }
}
```